src/EventTarget.ts

```diff
@@ -46,7 +46,8 @@ export class InternalEventTarget {
     const i = (this.events[type] as SplunkEventListener<T>[]).indexOf(listener);
 
     if (i >= 0) {
-      this.events[type].splice(i, 1);
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      this.events[type]!.splice(i, 1);
     }
   }
 
```

Comment:
Tbh I'd kinda wanna get rid of the extra payload layer here (eg currently the event object `{ payload: { sessionId: 'asd' } }` -> `{ sessionId: 'asd' }`)